src/go/rpk/pkg/config/config.go

```diff
@@ -46,16 +46,16 @@ type Config struct {
 }
 
 type RedpandaConfig struct {
-	Directory          string        `yaml:"data_directory" json:"dataDirectory"`
-	RPCServer          SocketAddress `yaml:"rpc_server" json:"rpcServer"`
-	AdvertisedRPCAPI   SocketAddress `yaml:"advertised_rpc_api" json:"advertisedRpcApi"`
-	KafkaApi           SocketAddress `yaml:"kafka_api" json:"kafkaApi"`
-	AdvertisedKafkaApi SocketAddress `yaml:"advertised_kafka_api" json:"advertisedKafkaApi"`
-	KafkaApiTLS        ServerTLS     `yaml:"kafka_api_tls" json:"kafkaApiTls"`
-	AdminApi           SocketAddress `yaml:"admin" json:"admin"`
-	Id                 int           `yaml:"node_id" json:"id"`
-	SeedServers        []*SeedServer `yaml:"seed_servers" json:"seedServers"`
-	DeveloperMode      bool          `yaml:"developer_mode" json:"developerMode"`
+	Directory          string         `yaml:"data_directory" json:"dataDirectory"`
+	RPCServer          SocketAddress  `yaml:"rpc_server" json:"rpcServer"`
+	AdvertisedRPCAPI   *SocketAddress `yaml:"advertised_rpc_api,omitempty" json:"advertisedRpcApi"`
+	KafkaApi           SocketAddress  `yaml:"kafka_api" json:"kafkaApi"`
```

Comment:
why not be consistent and make all socket addresses a pointer ?

Reply:
Because not all of them are optional.